src/tracing/esp32_trace/esp32_tracing.cpp

```diff
@@ -26,6 +26,113 @@
 namespace chip {
 namespace Tracing {
 namespace Insights {
+namespace {
+static constexpr size_t gPermitListSize = 10;
+struct PermitListEntry
+{
+    size_t hash;
+};
+
+// Implements a murmurhash with 0 seed.
+static size_t MurmurHash(const void * key)
+{
+    const size_t multiplier    = 0x5bd1e995;
+    const size_t shift         = 24;
+    const unsigned char * data = (const unsigned char *) key;
+    size_t hash                = 0;
+
+    while (*data)
+    {
+        size_t value = *data++;
+        value *= multiplier;
+        value ^= value >> shift;
+        value *= multiplier;
+        hash *= multiplier;
+        hash ^= value;
+    }
+
+    hash ^= hash >> 13;
+    hash *= multiplier;
+    hash ^= hash >> 15;
+
+    return hash;
+}
+
+/* PASESession,CASESession,NetworkCommissioning,GeneralCommissioning,OperationalCredentials
+ * are well known permitted entries.
+ */
+static PermitListEntry gPermitList[gPermitListSize] = {
+    { MurmurHash("PASESession") },
+    { MurmurHash("CASESession") },
+    { MurmurHash("NetworkCommissioning") },
+    { MurmurHash("GeneralCommissioning") },
+    { MurmurHash("OperationalCredentials") },
+};
+
+static bool IsPermitted(size_t hashValue)
+{
+    for (size_t i = 0; i < gPermitListSize; i++)
+    {
+        if (gPermitList[i].hash == 0)
+        {
+            break;
+        }
+        if (hashValue == gPermitList[i].hash)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+} // namespace
+
+namespace ESP32Filter {
+
+CHIP_ERROR AddHashToPermitlist(const char * str)
+{
+    size_t hashValue = MurmurHash(str);
+    for (size_t i = 0; i < gPermitListSize; i++)
+    {
+        if (gPermitList[i].hash == 0)
+        {
+            gPermitList[i].hash = hashValue;
+            return CHIP_NO_ERROR;
+        }
+        if (hashValue == gPermitList[i].hash)
+        {
+            return CHIP_ERROR_INCORRECT_STATE;
+        }
+    }
+    return CHIP_ERROR_NO_MEMORY;
+}
+
+void RemoveHashFromPermitlist(const char * str)
+{
+    size_t hashValue = MurmurHash(str);
+    size_t i         = 0;
+    while (i < gPermitListSize)
```

Comment:
this could maybe be:

```cpp
auto *end = gPermitList + kPermitListMaxSize;
std::fill(std::remove(gPermitList, end, hashValue), end, 0);
```

Saves a bit of vertical space....